openedx/core/djangoapps/oauth_dispatch/tests/test_jwt.py

```diff
@@ -79,6 +79,20 @@ def test_dot_create_jwt_for_token_forced_asymmetric(self):
         jwt_token = self._create_jwt_for_token(DOTAdapter(), use_asymmetric_key=False)
         self._assert_jwt_is_valid(jwt_token, should_be_asymmetric_key=True)
 
+    def test_kid_not_in_jwt_header(self):
```

Comment:
You should have 4 tests. Asymmetric True/False combined with `JWT_AUTH_ADD_KID_HEADER` True/False. I'm guessing that the Asymmetric False and `JWT_AUTH_ADD_KID_HEADER` True case should fail, which you then can fix according to my earlier comment. We'll see.

Reply:
Thanks for adding the other tests. 

I see this failure:
> FAILED openedx/core/djangoapps/oauth_dispatch/tests/test_jwt.py::TestCreateJWTs::test_kid_in_jwt_header_with_asymmetric_key_and_kid_enabled - AttributeError: module 'openedx.core.djangoapps.oauth_dispatch.jwt' has no attribute 'get_unverified_header'

You should be able to run the following in an lms shell in devstack to run your test locally:
```
pytest openedx/core/djangoapps/oauth_dispatch/tests/test_jwt.py::TestCreateJWTs::test_kid_in_jwt_header_with_asymmetric_key_and_kid_enabled
```